solr/core/src/java/org/apache/solr/packagemanager/PackageUtils.java

```diff
@@ -268,17 +270,8 @@ public static void print(String color, Object message) {
   }
 
   public static String[] validateCollections(String collections[]) {
```

Comment:
out of curisioty, is this method `validateCollections` called from multiple places?  If not, maybe best to just eliminate it all together ;-)

Reply:
Only in one other file: [PackageTool deploy](https://github.com/apache/solr/blob/main/solr/core/src/java/org/apache/solr/cli/PackageTool.java#L184) and [PackageTool undeploy](https://github.com/apache/solr/blob/main/solr/core/src/java/org/apache/solr/cli/PackageTool.java#L214).

We could remove this method entirely and use something like this inline:

```java
Arrays.stream(cli.getOptionValue("collections").split(","))
    .map(SolrIdentifierValidator::validateCollectionName)
    .toArray(String[]::new)
```

in both locations, although I sincerely hope there's a slightly shorter approach I've not thought of yet!

We could do something a bit less packed together given that `String[] collections` isn't final, but that'd take a bit more work to make sure it doesn't get a bit lengthy or involve multiple short-lived arrays.

Reply:
Actually, since we have a few places we use it, I think I like the explicit method...   Sometimes these stream style metods look a bit too clever!

Reply:
I can't tell you how badly I wish we had C# LINQ style helpers to just apply to Iterables - something like `.split(/**/).map(/**/).toArray()` would have been **so** much easier to read 😅